File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/relationshipsvb/vb/workwithrelationships3.vb
' <snippetworkwithrelationships3>


 ''' <summary>
 ''' Determines whether the entity can participate in a many-to-many relationship.
 ''' </summary>
 ''' <param name="entity">Entity</param>
 ''' <returns></returns>
 Public Function EligibleCreateManyToManyRelationship(ByVal entity As String) As Boolean
  Dim canManyToManyRequest As CanManyToManyRequest =
   New CanManyToManyRequest With {.EntityName = entity}

  Dim canManyToManyResponse As CanManyToManyResponse =
   CType(_serviceProxy.Execute(canManyToManyRequest), CanManyToManyResponse)

  If Not canManyToManyResponse.CanManyToMany Then
   Console.WriteLine("Entity {0} can't participate in a many-to-many relationship.", entity)
  End If

  Return canManyToManyResponse.CanManyToMany
 End Function

' </snippetworkwithrelationships3>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/relationships/cs/workwithrelationships3.cs
// <snippetworkwithrelationships3>


 /// <summary>
 /// Determines whether the entity can participate in a many-to-many relationship.
 /// </summary>
 /// <param name="entity">Entity</param>
 /// <returns></returns>
 public bool EligibleCreateManyToManyRelationship(string entity)
 {
     CanManyToManyRequest canManyToManyRequest = new CanManyToManyRequest
     {
         EntityName = entity
     };

     CanManyToManyResponse canManyToManyResponse =
         (CanManyToManyResponse)_serviceProxy.Execute(canManyToManyRequest);

     if (!canManyToManyResponse.CanManyToMany)
     {
         Console.WriteLine(
             "Entity {0} can't participate in a many-to-many relationship.", 
             entity);
     }

     return canManyToManyResponse.CanManyToMany;
 }

// </snippetworkwithrelationships3>